digitalocean/datasource_digitalocean_droplet.go

```diff
@@ -174,7 +179,15 @@ func dataSourceDigitalOceanDropletRead(d *schema.ResourceData, meta interface{})
 		opts.Page = page + 1
 	}
 
-	if v, ok := d.GetOk("tag"); ok {
+	if v, ok := d.GetOk("id"); ok {
+		droplet, err := findDropletById(dropletList, v.(int))
```

Comment:
While Droplet names are not guaranteed to be unique, their IDs are. So there is no need to loop through them all if we have the ID. We can just get it using the ID:

https://godoc.org/github.com/digitalocean/godo#DropletsServiceOp.Get

Reply:
That makes sense; then the ID lookup would short-circuit the need to even do the earlier `for` loop. I'll make the change.

Reply:
Change made.